weave/flow/eval.py

```diff
@@ -222,7 +223,27 @@ async def predict_and_score(self, model: Union[Op, Model], example: dict) -> dic
         scorers = self._post_init_scorers
 
         for scorer in scorers:
-            apply_scorer_result = await model_call.apply_scorer(scorer, example)
+            if isinstance(scorer, PairwiseScorer):
```

Comment:
This seems wrong to couple the logic here. I would have expected the implementation of `PairwiseScorer` to handle this internally